### PR TITLE
[feat] more accurate retry policy (+cosmetic refactor +tests refactor +change defaults)

### PR DESCRIPTION
**Pull request by guysegal**

### CODE REVIEW

**Comment by guysegal** [comment]
@danielwitz ready for review

File: src/main/java/configuration/Config.java [hunk 1]
```diff
@@ -112,11 +112,10 @@ public static void init() throws Exception {
             );
         }
 
-        RETRY_PROCESS_WHEN_STATUS_CODE_MATCH =
-            getOptionalString(dotenv, "RETRY_PROCESS_WHEN_STATUS_CODE_MATCH", "5[0-9][0-9]");
+        RETRY_PROCESS_WHEN_STATUS_CODE_MATCH = getOptionalString(dotenv, "RETRY_PROCESS_WHEN_STATUS_CODE_MATCH", "500");
 
         PRODUCE_TO_RETRY_TOPIC_WHEN_STATUS_CODE_MATCH =
-            getOptionalString(dotenv, "PRODUCE_TO_RETRY_TOPIC_WHEN_STATUS_CODE_MATCH", "408");
+            getOptionalString(dotenv, "PRODUCE_TO_RETRY_TOPIC_WHEN_STATUS_CODE_MATCH", "500|503|504|408");
```
Comment (danielwitz) [inline src/main/java/configuration/Config.java:118]: what about 502?

File: tests/specs/produceToDeadLetterTopicWhenValueIsNotValidJson.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        await orchestrator.stop();
+    });
+
+    it('consumer should produce to dead letter topic when value is not valid JSON', async () => {
+        await mockHttpTarget(orchestrator.wiremockClient, '/consume', 200);
+
+        //act
+        await produce(orchestrator, {
+            topic: 'foo',
+            messages: [{value: 'wat?'}],
```
Comment (danielwitz) [inline tests/specs/produceToDeadLetterTopicWhenValueIsNotValidJson.spec.ts:41]: why is that not a valid json?

File: tests/specs/consumeBursrsOfRecords.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import type {Orchestrator} from '../testcontainers/orchestrator.js';
```
Comment (danielwitz) [inline tests/specs/consumeBursrsOfRecords.spec.ts:1]: file's name is wrong

File: src/main/java/target/TargetRetryPolicy.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            .withBackoff(
+                connectionFailureDelay,
+                connectionFailureMaxDelay,
+                ChronoUnit.MILLIS,
+                connectionFailureDelayFactor
+            )
+            .withMaxDuration(connectionFailureMaxDuration)
+            .withMaxAttempts(Config.CONNECTION_FAILURE_RETRY_POLICY_MAX_RETRIES)
+            .handle(IOException.class)
+            .handleResultIf(r -> Integer.toString(r.code()).matches("503"))
```
Comment (danielwitz) [inline src/main/java/target/TargetRetryPolicy.java:38]: 502,503,504

**Review by danielwitz — APPROVED** [review]

File: tests/specs/consumeBursrsOfRecords.spec.ts (hunk 3 above)
Comment (guysegal) [reply tests/specs/consumeBursrsOfRecords.spec.ts]: fixed

File: src/main/java/configuration/Config.java (hunk 1 above)
Comment (guysegal) [reply src/main/java/configuration/Config.java]: added

**Pull request merged**